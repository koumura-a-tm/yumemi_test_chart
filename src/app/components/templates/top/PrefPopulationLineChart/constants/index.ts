import { DashStyleValue } from 'highcharts'

export const options = [
  { label: '総人口', value: '総人口' },
  { label: '年少人口', value: '年少人口' },
  { label: '生産年齢人口', value: '生産年齢人口' },
  { label: '老年人口', value: '老年人口' },
]

export const symbols = [
  'circle',
  'square',
  'diamond',
  'triangle',
  'triangle-down',
]

export const dashStyles: DashStyleValue[] = ['Solid', 'LongDash', 'ShortDot']

export const colors = [
  '#ff4b00',
  '#03af7a',
  '#005aff',
  '#4dc4ff',
  '#ff8082',
  '#f6aa00',
  '#990099',
  '#804000',
]
