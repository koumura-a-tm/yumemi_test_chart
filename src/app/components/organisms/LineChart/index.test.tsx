import { render } from '@testing-library/react'
import { LineChart } from '.'
import { options, optionsEmpty } from './constants'

describe('LineChart', () => {
  test('Snapshot: データが空の場合', () => {
    const { container } = render(<LineChart options={optionsEmpty} />)
    expect(container).toMatchSnapshot()
  })

  test('Snapshot: データが存在する場合', () => {
    const { container } = render(<LineChart options={options} />)
    expect(container).toMatchSnapshot()
  })
})
