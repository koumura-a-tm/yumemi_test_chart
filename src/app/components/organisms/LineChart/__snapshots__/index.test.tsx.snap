// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart Snapshot: データが存在する場合 1`] = `
<div>
  <div
    data-highcharts-chart="1"
    style="overflow: hidden; pointer-events: auto;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-2"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
    >
      <svg
        aria-label="U.S Solar Employment Growth"
        class="highcharts-root"
        height="400"
        role="img"
        style="font-family: Helvetica, Arial, sans-serif; font-size: 1rem;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 11.4.6
        </desc>
        <defs>
          <filter
            id="highcharts-drop-shadow-1"
          >
            <fedropshadow
              dx="1"
              dy="1"
              flood-color="#000000"
              flood-opacity="0.75"
              stdDeviation="2.5"
            />
          </filter>
          <clippath
            id="highcharts-14-"
          >
            <rect
              fill="none"
              height="375"
              width="NaNNaNNaNNaNNaN"
              x="0"
              y="0"
            />
          </clippath>
          <clippath
            id="highcharts-15-"
          >
            <rect
              class="highcharts-animating"
              fill="none"
              height="375"
              width="NaN"
              x="0"
              y="0"
            />
          </clippath>
          <clippath
            id="highcharts-16-"
          >
            <rect
              fill="none"
              height="574"
              width="NaN"
              x="-99"
              y="-99"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          filter="none"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          filter="none"
          height="375"
          width="NaN"
          x="NaN"
          y="10"
        />
        <g
          class="highcharts-grid highcharts-xaxis-grid"
          data-z-index="1"
        />
        <g
          class="highcharts-grid highcharts-yaxis-grid"
          data-z-index="1"
        />
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="375"
          stroke="#cccccc"
          stroke-width="0"
          width="0"
          x="0"
          y="10"
        />
        <g
          class="highcharts-axis highcharts-xaxis"
          data-z-index="2"
        >
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
            stroke="#333333"
            stroke-width="1"
          />
        </g>
        <g
          class="highcharts-axis highcharts-yaxis"
          data-z-index="2"
        >
          <text
            class="highcharts-axis-title"
            data-z-index="7"
            style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
            text-anchor="middle"
            transform="translate(0,0) rotate(270 NaN 197.5)"
            x="NaN"
            y="197.5"
          >
            Number of Employees
          </text>
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
            stroke="#333333"
            stroke-width="0"
          />
        </g>
        <g
          class="highcharts-series-group"
          data-z-index="3"
          filter="none"
        >
          <g
            class="highcharts-series highcharts-series-0 highcharts-line-series highcharts-color-0"
            clip-path="url(#highcharts-15-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-0 highcharts-line-series highcharts-color-0"
            clip-path="url(#highcharts-16-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-series highcharts-series-1 highcharts-line-series highcharts-color-1"
            clip-path="url(#highcharts-15-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-1 highcharts-line-series highcharts-color-1"
            clip-path="url(#highcharts-16-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-series highcharts-series-2 highcharts-line-series highcharts-color-2"
            clip-path="url(#highcharts-15-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-2 highcharts-line-series highcharts-color-2"
            clip-path="url(#highcharts-16-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-series highcharts-series-3 highcharts-line-series highcharts-color-3"
            clip-path="url(#highcharts-15-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-3 highcharts-line-series highcharts-color-3"
            clip-path="url(#highcharts-16-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-series highcharts-series-4 highcharts-line-series highcharts-color-4"
            clip-path="url(#highcharts-15-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
          <g
            class="highcharts-markers highcharts-series-4 highcharts-line-series highcharts-color-4"
            clip-path="url(#highcharts-16-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(NaN,10) scale(1 1)"
          />
        </g>
        <text
          class="highcharts-title"
          data-z-index="4"
          style="font-size: 1.2em; color: rgb(51, 51, 51); font-weight: bold; fill: #333333;"
          text-anchor="start"
          x="10"
          y="10"
        >
          U.S Solar Employment Growth
        </text>
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
          text-anchor="start"
          x="10"
          y="10"
        >
          By Job Category. Source: 
          <a
            href="https://irecusa.org/programs/solar-jobs-census/"
            target="_blank"
          >
            <tspan>
              IREC
            </tspan>
          </a>
          .
        </text>
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
          text-anchor="start"
          x="10"
          y="387"
        />
        <g
          class="highcharts-legend highcharts-no-tooltip"
          data-z-index="7"
          transform="translate(590,175)"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            filter="none"
            rx="0"
            ry="0"
            stroke="#999999"
            stroke-width="0"
          />
          <g
            data-z-index="1"
          >
            <g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-0 highcharts-series-0"
                data-z-index="1"
                transform="translate(8,3)"
              >
                <path
                  class="highcharts-graph"
                  d="M 1 5 L 15 5"
                  fill="none"
                  stroke="#2caffe"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 5 A 0 0 0 1 1 8 5 Z"
                  fill="#2caffe"
                  opacity="1"
                  stroke="#ffffff"
                  stroke-width="0"
                />
                <text
                  data-z-index="2"
                  style="color: rgb(51, 51, 51); cursor: pointer; font-size: 0.8em; text-decoration: none; fill: #333333;"
                  text-anchor="start"
                  x="21"
                  y="5"
                >
                  Installation & Developers
                </text>
              </g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-1 highcharts-series-1"
                data-z-index="1"
                transform="translate(8,10)"
              >
                <path
                  class="highcharts-graph"
                  d="M 1 5 L 15 5"
                  fill="none"
                  stroke="#544fc5"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 5 L 8 5 L 8 5 L 8 5 Z"
                  fill="#544fc5"
                  opacity="1"
                  stroke="#ffffff"
                  stroke-width="0"
                />
                <text
                  data-z-index="2"
                  style="color: rgb(51, 51, 51); cursor: pointer; font-size: 0.8em; text-decoration: none; fill: #333333;"
                  text-anchor="start"
                  x="21"
                  y="5"
                >
                  Manufacturing
                </text>
              </g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-2 highcharts-series-2"
                data-z-index="1"
                transform="translate(8,17)"
              >
                <path
                  class="highcharts-graph"
                  d="M 1 5 L 15 5"
                  fill="none"
                  stroke="#00e272"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 5 L 8 5 L 8 5 L 8 5 Z"
                  fill="#00e272"
                  opacity="1"
                  stroke="#ffffff"
                  stroke-width="0"
                />
                <text
                  data-z-index="2"
                  style="color: rgb(51, 51, 51); cursor: pointer; font-size: 0.8em; text-decoration: none; fill: #333333;"
                  text-anchor="start"
                  x="21"
                  y="5"
                >
                  Sales & Distribution
                </text>
              </g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-3 highcharts-series-3"
                data-z-index="1"
                transform="translate(8,24)"
              >
                <path
                  class="highcharts-graph"
                  d="M 1 5 L 15 5"
                  fill="none"
                  stroke="#fe6a35"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 5 L 8 5 L 8 5 Z"
                  fill="#fe6a35"
                  opacity="1"
                  stroke="#ffffff"
                  stroke-width="0"
                />
                <text
                  data-z-index="2"
                  style="color: rgb(51, 51, 51); cursor: pointer; font-size: 0.8em; text-decoration: none; fill: #333333;"
                  text-anchor="start"
                  x="21"
                  y="5"
                >
                  Operations & Maintenance
                </text>
              </g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-4 highcharts-series-4"
                data-z-index="1"
                transform="translate(8,31)"
              >
                <path
                  class="highcharts-graph"
                  d="M 1 5 L 15 5"
                  fill="none"
                  stroke="#6b8abc"
                  stroke-linecap="round"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 5 L 8 5 L 8 5 Z"
                  fill="#6b8abc"
                  opacity="1"
                  stroke="#ffffff"
                  stroke-width="0"
                />
                <text
                  data-z-index="2"
                  style="color: rgb(51, 51, 51); cursor: pointer; font-size: 0.8em; text-decoration: none; fill: #333333;"
                  text-anchor="start"
                  x="21"
                  y="5"
                >
                  Other
                </text>
              </g>
            </g>
          </g>
        </g>
        <g
          class="highcharts-axis-labels highcharts-xaxis-labels"
          data-z-index="7"
        >
          <text
            style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
            x="0"
          >
            x
          </text>
        </g>
        <g
          class="highcharts-axis-labels highcharts-yaxis-labels"
          data-z-index="7"
        />
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor: pointer; color: rgb(153, 153, 153); font-size: 0.6em; fill: #999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
</div>
`;

exports[`LineChart Snapshot: データが空の場合 1`] = `
<div>
  <div
    data-highcharts-chart="0"
    style="overflow: hidden; pointer-events: auto;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
    >
      <svg
        aria-label="U.S Solar Employment Growth"
        class="highcharts-root"
        height="400"
        role="img"
        style="font-family: Helvetica, Arial, sans-serif; font-size: 1rem;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 11.4.6
        </desc>
        <defs>
          <filter
            id="highcharts-drop-shadow-0"
          >
            <fedropshadow
              dx="1"
              dy="1"
              flood-color="#000000"
              flood-opacity="0.75"
              stdDeviation="2.5"
            />
          </filter>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          filter="none"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          filter="none"
          height="375"
          width="580"
          x="10"
          y="10"
        />
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="375"
          stroke="#cccccc"
          stroke-width="0"
          width="580"
          x="10"
          y="10"
        />
        <g
          class="highcharts-series-group"
          data-z-index="3"
          filter="none"
        />
        <text
          class="highcharts-title"
          data-z-index="4"
          style="font-size: 1.2em; color: rgb(51, 51, 51); font-weight: bold; fill: #333333;"
          text-anchor="start"
          x="10"
          y="10"
        >
          U.S Solar Employment Growth
        </text>
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
          text-anchor="start"
          x="10"
          y="10"
        >
          By Job Category. Source: 
          <a
            href="https://irecusa.org/programs/solar-jobs-census/"
            target="_blank"
          >
            <tspan>
              IREC
            </tspan>
          </a>
          .
        </text>
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
          text-anchor="start"
          x="10"
          y="387"
        />
        <g
          class="highcharts-legend highcharts-no-tooltip"
          data-z-index="7"
          visibility="hidden"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            filter="none"
            height="8"
            rx="0"
            ry="0"
            stroke="#999999"
            stroke-width="0"
            width="8"
            x="0"
            y="0"
          />
          <g
            data-z-index="1"
          >
            <g />
          </g>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor: pointer; color: rgb(153, 153, 153); font-size: 0.6em; fill: #999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
</div>
`;
